### PR TITLE
test:add auction liquidation cases

### PR DESCRIPTION
**Pull request by alexliparallel**
## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

**Comment by yrong** [comment]
image

@alexliparallel  seems task in the sheet about this one is a little confusing, did not notice you've picked it.  So In this [pr](https://github.com/para-space/paraspace-core/pull/160) I've  also done some work and already merged. Could you check with latest code if anything important missing? We can merge or just use your commit if necessary

**Comment by alexliparallel** [comment]
@yrong OK,Let's Review it together

File: test-suites/_pool_core_erc721_auction_liquidation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,918 @@
...
+      // start auction again
+      await expect(
+        pool
+          .connect(liquidator.signer)
+          .startAuction(borrower.address, bayc.address, 0)
+      ).to.be.revertedWith(ProtocolErrors.AUCTION_ALREADY_STARTED);
+    });
+  });
+
+  describe("Do not revert to snapshot on every step", () => {
```
Comment (yrong) [inline test-suites/_pool_core_erc721_auction_liquidation.spec.ts:834]: suggest we merge this with revert and keep every use-case atomic and independent with each other

**Comment by yrong** [comment]
@GopherJ  this pr can be merged

**Review by GopherJ — APPROVED** [review]

**Pull request merged**